### PR TITLE
MERC-5943 Surface EA error fields and track bid/ask violations

### PR DESCRIPTION
**Pull request by martin-cll**
### Context

Per @mjk90, EAs will track if the bid/ask invariant is violated and will send through a specific error message with a 500 status if so:
```
{
    "status": "errored",
    "statusCode": 500,
    "error": {
        "name": "AdapterLWBAError",
        "message": "Invariant violation. Mid price must be between bid and ask prices. Got: (mid: 3051.493596362398, bid: 3051.360434937159, ask: 3051.4270156497787)"
    }
}
```

We want to capture the EA's error, propagate it through the pipeline, catch it in the Mercury observer, and track it. We also want to make sure that the observation is not used in producing a report.

### Changes

The EA is called in the bridge task. Currently, for a 500 response, the `makeHTTPRequest` helper returns a generic error which isn't the most helpful. I've added a `eautils.BestEffortExtractEAError` helper to build a structured EA error that captures the error name and message.

Then, in the Mercury observer, we can look for this error (it will be a wrapped error from parsing one of the pipeline's 3 pricing fields). If this error is found then we will send an enhanced telemetry record for it.

Note that previously we would not send telemetry if there is any error so this is new behaviour cc @mjk90. I've also added a new `EnhancedEAMercury.dp_invariant_violation_detected` protobuf field to track this.

### CODE REVIEW

**Review by mjk90 — COMMENTED** [review]
This looks good to me generally but we should also get approval from someone with more core experience. The EA framework has now been updated with a more specific error type to make it easier to pick up in the node:
```
{
    "status": "errored",
    "statusCode": 500,
    "error": {
        "name": "AdapterLWBAError",
        "message": "Invariant violation. LWBA response must contain mid, bid and ask prices. Got: (mid: 3064.866547022821, bid: 3064.8327375532613, ask: null)"
    }
}
```

**Review by samsondav — CHANGES_REQUESTED** [review]
LGTM just need a regression test

File: core/services/relay/evm/mercury/v3/data_source.go
Comment (mjk90) [inline core/services/relay/evm/mercury/v3/data_source.go]: Could we send this in EA telemetry instead of just recording a node metric?
https://github.com/smartcontractkit/chainlink/blob/develop/core/services/ocrcommon/telemetry.go

File: core/services/pipeline/common_http.go
Comment (martin-cll) [inline core/services/pipeline/common_http.go:67]: This helper is only used by the http and bridge tasks and in both cases it's fine to return the response on error.

**Review by mjk90 — COMMENTED** [review]
The new telemetry change looks good to me but I'll leave it to @samsondav to give the final approval 👍

**Review by samsondav — APPROVED** [review]

**Review by mjk90 — APPROVED** [review]

**Pull request merged**
